test/mapRoutes.js

```diff
@@ -0,0 +1,189 @@
+const { it, describe } = require('mocha')
+const Router = require('..')
+const utils = require('./support/utils')
+
+const assert = utils.assert
+
+describe('mapRoutes', function () {
+  it('should return empty array for router with no registered routes', function () {
+    const router = new Router()
+
+    assert.deepStrictEqual(router.mapRoutes(), [])
+  })
+
+  it('should map different route types including strings, regex patterns, and parameter routes', function () {
+    const router = new Router()
+
+    router.all('/', noop)
+    router.route('/test2/')
+    router.route('/test/').get(noop)
+    // With regex patterns path-to-regexp fail
+    // router.all(/^\/[a-z]oo$/, noop)
+    router.get(['/foo', '/bar'], noop)
+    router.post('/:id/setting/:thing', noop)
+
+    assert.deepStrictEqual(router.mapRoutes(),
+      [
+        { path: '/', methods: ['_ALL'], keys: [], options: { strict: undefined, caseSensitive: undefined } },
```

Comment:
Ah, I didn't notice this magic sting above, but I am not a fan of `_ALL`. We should list them here.